consensus/src/lib.rs

```diff
@@ -32,6 +32,8 @@ use common::{
     primitives::{BlockHeight, Id},
 };
 
+const POW_MAX_NONCE: u128 = 10_000;
```

Comment:
Wrong. There should not be a limit.

Reply:
We talked about this the other day, saying that we should have a limit so that we're not unnecessarily checking `stop_flag` in each iteration.

Also, now that I think about it, if we never stop unless nonce exhaustion or the stop flag is set, then `block_header`'s timestamp will be fixed the whole time